Utilities/ocvalidate/ValidateKernel.c

```diff
@@ -218,38 +218,40 @@ CheckKernelAdd (
         OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH)
         ));
       ++ErrorCount;
-    }
-
-    //
-    // There is one missing '\\' after the concatenation of BundlePath and ExecutablePath. Append one.
-    //
-    ExecutableFixedSize   = L_STR_LEN (OPEN_CORE_KEXT_PATH) + AsciiStrLen (BundlePath) + 1;
-    ExecutablePathSumSize = ExecutableFixedSize + AsciiStrSize (ExecutablePath);
-    if (ExecutablePathSumSize > OC_STORAGE_SAFE_PATH_MAX) {
-      DEBUG ((
-        DEBUG_WARN,
-        "Kernel->Add[%u]->ExecutablePath (length %u) is too long (should not exceed %u)!\n",
-        Index,
-        AsciiStrLen (ExecutablePath),
-        OC_STORAGE_SAFE_PATH_MAX - ExecutableFixedSize
-        ));
-      ++ErrorCount;
-    }
+    } else {
+      //
+      // Only check the ExecutablePath and PlistPath if BundlePath did not trigger an error.
```

Comment:
I believe it is nicer to display all the error messages, and your code display them fine without this 'else'. (Of course you are quite correct that the second and third error messages must be there, if the first one is.)